### PR TITLE
I dont want your damn citrus!

### PR DESCRIPTION
**Pull request by maximumpower55**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,6 +25,7 @@ dependencies {
 
 	// dependencies
 	modImplementation("org.quiltmc.quilted-fabric-api:quilted-fabric-api:$qfapi_version")
+	modImplementation("com.terraformersmc.terraform-api:terraform-wood-api-v1:$terraform_wood_api")
```
Comment (TropheusJ) [inline build.gradle:28]: add a QMJ dependency

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java [hunk 2]
```diff
@@ -115,6 +136,135 @@ public class PortalCubedBlocks {
 					}
 				}
 			});
+	// ----- lemon -----
+	public static final RotatedPillarBlock LEMON_LOG = REGISTRAR.blocks.create("lemon_log", RotatedPillarBlock::new)
+			.copyFrom(Blocks.OAK_LOG)
+			.settings(settings -> settings
+					.mapColor(MapColor.TERRACOTTA_GRAY)
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java:143]: this should not be on a separate line

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java [hunk 3]
```diff
@@ -115,6 +136,135 @@ public class PortalCubedBlocks {
...
+					.mapColor(MapColor.TERRACOTTA_YELLOW)
+			)
+			.build();
+	public static final PressurePlateBlock LEMON_PRESSURE_PLATE = REGISTRAR.blocks.create("lemon_pressure_plate", properties -> new PressurePlateBlock(BlockSetType.OAK, properties))
+			.copyFrom(Blocks.OAK_PRESSURE_PLATE)
+			.settings(settings -> settings
+					.mapColor(MapColor.TERRACOTTA_YELLOW)
+			)
+			.build();
+	private static final ResourceLocation LEMON_SIGN_TEXTURE = PortalCubed.id("entity/signs/lemon");
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java:218]: These shouldn't be stored here. also the name doesn't match the access

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedEntities.java [hunk 4]
```diff
@@ -24,6 +35,13 @@ public class PortalCubedEntities {
 			.renderer(() -> () -> PortalProjectileRenderer::new)
 			.build();
 
+	public static final EntityType<Lemonade> LEMONADE = REGISTRAR.entities.create("lemonade", Lemonade::create)
+			.configure(b -> b.maxChunkTrackingRange(4).trackingTickInterval(10))
+			.size(EntityDimensions.fixed(0.375f, 0.375f))
+			.renderer(() -> () -> ThrownItemRenderer::new)
+			.build();
+	public static final ResourceKey<TerraformBoatType> LEMON_BOAT = TerraformBoatTypeRegistry.createKey(PortalCubed.id("lemon"));
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedEntities.java:43]: this should be the TerraformBoatType itself, registered directly

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedEntities.java [hunk 5]
```diff
@@ -36,5 +54,17 @@ public class PortalCubedEntities {
...
+						.item(PortalCubedItems.LEMON_BOAT)
+						.chestItem(PortalCubedItems.LEMON_CHEST_BOAT)
+						.planks(PortalCubedBlocks.LEMON_PLANKS.asItem())
+						.build()
+		);
+	}
+
+	@ClientOnly
+	public static void initClient() {
+		TerraformBoatClientHelper.registerModelLayers(LEMON_BOAT.location(), false);
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedEntities.java:68]: just call this in client init

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedItems.java [hunk 6]
```diff
@@ -50,5 +78,15 @@ public class PortalCubedItems {
...
+		DispenserBlock.registerBehavior(LEMONADE, new LemonadeDispenseBehavior());
+		LootTableEvents.MODIFY.register((resourceManager, lootManager, id, tableBuilder, source) -> {
+			if (BuiltInLootTables.SNIFFER_DIGGING.equals(id) && source.isBuiltin())
+				tableBuilder.modifyPools(builder -> builder.add(LootItem.lootTableItem(PortalCubedBlocks.LEMON_SAPLING)));
+		});
+	}
+
+	@ClientOnly
+	public static void initClient() {
+		ItemProperties.register(LEMONADE, PortalCubed.id("armed"), (stack, level, entity, i) -> LemonadeItem.isArmed(stack) ? 1 : 0);
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedItems.java:90]: this should just be in client init

File: src/main/java/io/github/fusionflux/portalcubed/content/misc/Lemonade.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.projectile.ThrowableItemProjectile;
+import net.minecraft.world.item.Item;
+
+import net.minecraft.world.level.Level;
+
+import org.jetbrains.annotations.NotNull;
+
+public class Lemonade extends ThrowableItemProjectile {
+	public static final int TICKS_PER_TICK = 20;
```
Comment (TropheusJ) [inline src/main/java/io/github/fusionflux/portalcubed/content/misc/Lemonade.java:19]: what

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java (hunk 2 above)
Comment (maximumpower55) [reply src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java:143]: Blame carter :ioa:

File: src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java (hunk 3 above)
Comment (maximumpower55) [reply src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java:218]: Where should they be stored

Comment (TropheusJ) [reply src/main/java/io/github/fusionflux/portalcubed/content/PortalCubedBlocks.java:199]: I guess it's fine. make them public though

**Review by TropheusJ — APPROVED** [review]

**Pull request merged**